src/internal/serviceenv/config.go

```diff
@@ -30,12 +30,13 @@ type GlobalConfiguration struct {
 	PostgresServiceSSL  string `env:"POSTGRES_SERVICE_SSL,default=disable"`
 	PostgresDBName      string `env:"POSTGRES_DATABASE_NAME"`
 
-	// PPSSpecCommitID is only set for workers and sidecar pachd instances.
-	// Because both pachd and worker need to know the spec commit (the worker so
-	// that it can avoid jobs for other versions of the same pipelines and the
-	// sidecar so that it can serve the S3 gateway) it's stored in the
+	// PPSSpecCommitID and PPSPipelineName are only set for workers and sidecar
```

Comment:
I'd love to do a follow-up to clean up these structs - I think this comment correctly captures the categories of config (global, pachd, worker, sidecar) but I don't think we've actually organized things that way?

Reply:
Given how spare the worker-specific struct is at this point, I wonder if it should just be called "PipelineSpecific", reclaim these variables, and get included in the sidecar and worker